classes/ActionScheduler_AdminView.php

```diff
@@ -35,9 +35,53 @@ public function init() {
 			} else {
 				add_action( 'admin_menu', array( $self, 'register_menu' ) );
 			}
+			add_action( 'admin_notices', array( $self, 'check_past_due_actions' ) );
 		}
 	}
 
+	/**
+	 * Check if there are any past due actions.
+	 *
+	 * This method will check if there are past due actions. Because Action Scheduler is action agnostic,
+	 * it will trigger an action with a list of hooks that callbacks can use to handle past due actions
+	 * as needed.
+	 *
+	 * To narrow down possible results, and improve query looking for past due actions, any plugin
+	 * must add their hook name to an array through the `action_scheduler_past_due_hooks`
+	 * filter.
+	 *
+	 * The action owner should be listening to the `action_schedule_past_due_action` action, 
+	 * and do something to warn the users about how to fix their WP-Cron or to run their actions manually.
+	 *
+	 * This method is executed on the `admin_notices`, so it is safe to render the admin notification.
+	 */
+	public function check_past_due_actions() {
+		$instance = ActionScheduler_Store::instance();
+
+		$actions = apply_filters( 'action_scheduler_past_due_hooks', array() );
```

Comment:
`$actions` is used later on to refer to an array of `ActionScheduler_Action` instances. It would be safe to avoid using the same variable name to refer to two different data sets in the one function. This variable would be better named `$action_hooks` or `$hooks` for that reason (and because that's more accurately the data type it contains).